cmd/kubeadm/app/preflight/checks.go

```diff
@@ -351,6 +356,38 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	return warns, nil
 }
 
+type KubernetesVersionCheck struct {
+	KubeadmVersion    string
+	KubernetesVersion string
+}
+
+func (kubever KubernetesVersionCheck) Check() (warnings, errors []error) {
+
+	// Skip this check for "super-custom builds", where apimachinery/the overall codebase version is not set.
+	if kubever.KubeadmVersion == "v0.0.0" {
+		return nil, nil
+	}
+
+	kadmVersion, err := semver.ParseTolerant(kubever.KubeadmVersion)
```

Comment:
Please use the `pkg/util/version` package instead of this one.
Unfortunate that `semver` is used here already, but leave that usage as-is for now.
Eventually we should not depend on `blang/semver`, but instead use what's provided in-tree for k8s (has more functionality/easier to update if necessary)